File: Ejemplos/Sincronismo/C++/atomic.cpp
#include <iostream>
#include <thread>
#include <vector>
#include <atomic>

std::atomic<int> count(0);

void Sum()
{
    for (int i=0; i<100; ++i) 
    {
        atomic_fetch_add(&count, 1);
    }
}

int main() 
{
    std::vector<std::thread> threads;

    for(int i=0; i<1000; ++i)
    {
        threads.push_back( std::thread(Sum) );
    }

    for(int i=0; i<1000; ++i)
    {
        threads[i].join();
    }

    std::cout<<"Resultado "<<count.load()<<std::endl;

    return EXIT_SUCCESS;
} 

File: Ejemplos/Sincronismo/Java/Atomic.java
import java.util.concurrent.atomic.AtomicInteger;

public class Atomic 
{
    static AtomicInteger count = new AtomicInteger(0);

    static class MyThread extends Thread 
    {
        public void run()
        {
            for (int i=0; i<100; ++i) 
            {
                count.getAndIncrement();
            }
        }
    }

    public static void main(String[] args) throws InterruptedException 
    {
        MyThread []threads = new MyThread[1000];
       
        for(int i=0; i<1000; ++i)
        {
            threads[i] = new MyThread();
        }

        for(int i=0; i<1000; ++i)
        {
            threads[i].start();
        }

        for(int i=0; i<1000; ++i)
        {
            threads[i].join();
        }

        System.out.println("Valor: " + count.get());		
    }
}
